datalad/distribution/publish.py

```diff
@@ -182,6 +192,33 @@ def _publish_data(ds, remote, paths, annex_copy_options, force, **kwargs):
     #     _log_push_info(ds.repo.push(remote=remote, refspec=['git-annex']))
 
 
+def _check_and_update_remote_server_info(ds, remote):
+    # if we managed to copy to "http" url  we should should try to trigger git
+    # update-server-info hook on the remote if there was ssh annexurl defined
+    # for it. Apparently we do that already in create_sibling ones, but here
+    # we need more checks and preparation
+    remote_url = ds.repo.config.get('remote.%s.url' % remote, None)
+    if remote_url:
+        remote_url = RI(remote_url)
+        if isinstance(remote_url, URL) and remote_url.scheme in (
+        'http', 'https'):
+            remote_annexurl = ds.repo.config.get('remote.%s.annexurl' % remote,
+                                                 None)
+            if remote_annexurl:
+                remote_annexurl_ri = RI(remote_annexurl)
+                if is_ssh(remote_annexurl_ri):
+                    ssh = ssh_manager.get_connection(remote_annexurl_ri)
+                    ssh('git -C {} update-server-info'.format(
+                        sh_quote(remote_annexurl_ri.path)))
+                    return True
```

Comment:
There are a few lines of piece of code in `create_sibling.py` which is doing a similar thing. Maybe we should consolidate right away, and not have two custom call implementations for `git update-server-info`.